pkg/registry/core/serviceaccount/strategy.go

```diff
@@ -42,7 +50,11 @@ func (strategy) NamespaceScoped() bool {
 }
 
 func (strategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
-	cleanSecretReferences(obj.(*api.ServiceAccount))
+	serviceAccount := obj.(*api.ServiceAccount)
+	cleanSecretReferences(serviceAccount)
+	if hasEnforceMountableSecretsAnnotation(obj.(*api.ServiceAccount)) {
+		warning.AddWarning(ctx, "", fmt.Sprintf("deprecated annotation %q on service account %s/%s", EnforceMountableSecretsAnnotation, serviceAccount.Namespace, serviceAccount.Name))
```

Comment:
We should link to the docs to help folks understand what do if the encounter this warning.

Reply:
I plan to push a PR to update this https://kubernetes.io/docs/concepts/configuration/secret/#configure-least-privilege-access-to-secrets to mention the deprecation and suggest using separate namespaces for isolation. I can update this warning to point to that link. wdyt?

Reply:
SGTM